core/src/Cuda/Kokkos_CudaSpace.cpp

```diff
@@ -172,6 +174,115 @@ void *CudaSpace::allocate(const char *arg_label, const size_t arg_alloc_size,
 }
 
 namespace {
+
+#ifndef CUDART_VERSION
+#error CUDART_VERSION undefined!
+#elif (defined(KOKKOS_ENABLE_IMPL_CUDA_MALLOC_ASYNC) && CUDART_VERSION >= 11020)
+
+/**
+ *  If a device has been initialized we will drop it in the set
+ * so we don't initialize it again
+ */
+std::unordered_set<int> _mem_pool_initialized;
+
+/**
+ * This is the initializer. It currently gets the device_id and stream
+ * that impl_alloc gets
+ *    @param device_id  The device id on which to alter the default mempool
+ *    @param stream     A specific stream for cudaMallocAsync
+ * 		@param error_code return any failure codes for cuda APIs
+ *    @return  				  true for success, false for failure
+ */
+bool initializeMempool(const int device_id, const cudaStream_t stream,
+                       cudaError_t &error_code) {
+  error_code = cudaSuccess;
+
+  std::cout << "Initializing Default Memory Pool for device " << device_id
+            << "\n";
```

Comment:
```suggestion;
```

Reply:
What is the comment here? It seems empty